pkg/products/monitoring/prometheusRules.go

```diff
@@ -0,0 +1,287 @@
+			},
+			Expr:   intstr.FromString("( predict_linear(kubelet_volume_stats_available_bytes{job='kubelet'}[1h], 4 * 3600) <= 0 AND kubelet_volume_stats_available_bytes{job='kubelet'} / kubelet_volume_stats_capacity_bytes{job='kubelet'} < 0.25)* on(namespace) group_left(label_monitoring_key) kube_namespace_labels{label_monitoring_key='middleware'}"),
+			For:    "15m",
+			Labels: map[string]string{"severity": "critical"},
+		},
+
+		{
+			Alert: fmt.Sprintf("PersistentVolumeErrors"),
+			Annotations: map[string]string{
+				"sop_url": "https://github.com/RHCloudServices/integreatly-help/blob/master/sops/2.x/alerts/pvc_storage.asciidoc#persistentvolumeerrors",
+				"message": "The PVC {{  $labels.persistentvolumeclaim  }} is in status {{  $labels.phase  }} in namespace {{  $labels.namespace }} ",
+			},
+			Expr:   intstr.FromString("(sum by(persistentvolumeclaim, namespace, phase) (kube_persistentvolumeclaim_status_phase{phase=~'Failed|Pending|Lost'}) * on ( namespace) group_left(label_monitoring_key) kube_namespace_labels{label_monitoring_key='middleware'}) > 0"),
+			For:    "15m",
+			Labels: map[string]string{"severity": "critical"},
+		}}
+
+	_, err := controllerutil.CreateOrUpdate(ctx, serverClient, rule, func() error {
+		rule.ObjectMeta.Labels = map[string]string{"integreatly": "yes", monitoringConfig.GetLabelSelectorKey(): monitoringConfig.GetLabelSelector()}
+		rule.Spec = monitoringv1.PrometheusRuleSpec{
+			Groups: []monitoringv1.RuleGroup{
+				{
+					Name:  "general.rules",
+					Rules: rules,
+				},
+			},
+		}
+		return nil
+	})
+
+	if err != nil {
+		logrus.Infof("Phase: %s reconcilePrometheusAlerts", err)
+
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("error creating backup PrometheusRule: %w", err)
+	}
+	return integreatlyv1alpha1.PhaseCompleted, nil
+}
+func (r *Reconciler) reconcileKubeStateMetricsMonitoringAlerts(ctx context.Context, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+	monitoringConfig := config.NewMonitoring(config.ProductConfig{})
+	rule := &monitoringv1.PrometheusRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ksm-monitoring-alerts",
+			Namespace: r.Config.GetOperatorNamespace(),
+		},
+	}
+
+	rules := []monitoringv1.Rule{
+		{
+			Alert: fmt.Sprintf("MiddlewareMonitoringPodCount"),
+			Annotations: map[string]string{
+				"sop_url": "https://github.com/RHCloudServices/integreatly-help/blob/master/sops/alerts_and_troubleshooting.md",
+				"message": "Pod count for namespace {{ $labels.namespace }} is {{ $value }}. Expected exactly 7 pods.",
+			},
+			Expr:   intstr.FromString("(1 - absent(kube_pod_status_ready{condition='true',namespace='redhat-rhmi-middleware-monitoring-operator'})) or sum(kube_pod_status_ready{condition='true',namespace='redhat-rhmi-middleware-monitoring-operator'}) != 7"),
+			For:    "5m",
+			Labels: map[string]string{"severity": "critical"},
+		}}
```

Comment:
```suggestion
		}
	}

```

^or else group the opening braces in a similar way

Reply:
if you separate them you get errors

Reply:
First thought: really? That's bizarre :thinking: 

Second thought: Oh, I think it's because there's a trailing comma missing after the element of the slice. Does this work better?

```suggestion
		},
	}

```